app/javascript/packs/contact-form-navigation-alert.js

```diff
@@ -0,0 +1,23 @@
+$(() => {
+   $("body").attr("data-turbolinks", false);
+
+   const formElement = $("form");
```

Comment:
May want to be more specific - if we have a `<form>` in the nav bar to switch roles, or something, it would catch that.

Reply:
> Hm, I was under the impression this could be simpler. Something like:
> 
> ```js
> let contactForm = document.querySelector('form#edit-contact, form#create-contact')
> let isContactFormDirty = false
> 
> if (contactForm) {
>   contactForm.addEventListener('change', () => isContactFormDirty = true)
>   contactForm.addEventListener('submit', () => isContactFormDirty = false)
> }
> 
> window.addEventListener('beforeunload', (event) => {
>   if (isContactFormDirty) {
>     e.preventDefault()
>     e.returnValue = ''
>   }
> })
> ```
> 
> Would that work?

The issue with the change event is that it will consider the form dirty even if the user restores the original values. Serialization will take the restored original values into account and the confirmation dialog won't be displayed.

Reply:
Got it. Note, though, that my question in this comment was actually about the specificity of the `'form'` selector, which I still think would be a bit of an issue.